### ISSUE TITLE
call reactor.wakeUp() in SIGCHLD handler

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7607]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#381|
|Type|defect|
|Created|2003-11-04 00:30:26Z|

<details><summary>Searchable metadata</summary>

```
trac-id__381 381
type__defect defect
reporter__itamarst itamarst
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1067905826000000 1067905826000000
changetime__1469240263507879 1469240263507879
version__ 
owner__ 
cc__itamarst
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
As per http://cr.yp.to/docs/selfpipe.html, "The SIGCHLD might go off while
select() is starting, too early to interrupt it, too late to change its
timeout." Thus, we need to wake the reactor so SIGCHLD is handled immediately.

This means Unix systems will always have a waker.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub commented|
|-|-|

In changeset 0538c2410148bf571dbf95ef125a15d42b685c08
```
#!CommitTicketReference repository="" revision="0538c2410148bf571dbf95ef125a15d42b685c08"
Merge pull request #381 from twisted/8646-rodrigc-noclasstype

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8646

Use _OldStyleClass and _OldStyleInstance for ClassType and InstanceType
```
